x8.md

```diff
@@ -41,160 +45,29 @@ As you can see, the message is an array and each index value has its meaning:
-2. Participants of a given Whisper network in order to communicate with each other MUST accept messages with lowered PoW value. More in (Whisper node configuration)(#whisper-node-configuration).
-3. Time MUST be synced between all nodes participating in the given network (this is intrinsic requirement of the Whisper specification as well). A clock drift between two peers larger than 20 seconds MAY result in discarding incoming messages.
-
-# Protocol Overview
-
-Notice: this protocol is documented post factum. The goal of it is to clearly present the current design and prepare the ground for its second version.
-
-The implementation of this protocol is mainly done in https://github.com/status-im/status-react and https://github.com/status-im/status-go repositories.
-
-The goal of this protocol is to allow people running Ethereum nodes with Whisper service enabled to exchange messages that are end-to-end encrypted in a way that guarantees [darkness to some extent](https://github.com/ethereum/go-ethereum/wiki/Achieving-Darkness).
-
-It's important to notice that messages [are not limited to be text messages](#content-types) only. They can also have special meaning depending on the client implementation. For example, in the current implementation, there are message which informs about Eth requests.
-
-This protocol consist of three components:
-* payload
-* Whisper adapter
-* offline messaging.
-
-[The payload section](#payload) describes how the messages are encoded and decoded and what each fields of a message means. This is required to properly interpret messages by the client.
-
-Whisper adapter specifies interaction with the Whisper service with regards to keys management, configuration and attaching metadata required to properly forward and process messages.
-
-Offline messaging describes how the protocol handles delivering messages when one or more participants are offline and the messages expire in the Whisper network.
-
-The protocol does not specify additional things like peers discovery, running Whisper nodes, underlying p2p protocols etc.
-
-# Payload
-
-Payload is an array of bytes exchanged between peers. In the case of this protocol, the payload is encoded using [transit format](https://github.com/cognitect/transit-format). It is a text-based format so even encoded version is easily readable by humans.
-
-Payload contains the following fields:
-1. text `string`
-2. content type `enum { text/plain }` (more in [Content types](#content-types))
-3. message type `enum { public-group-user-message, user-message, group-user-message }` (more in [Message types](#message-types))
-4. clock `int64`
-5. timestamp `int64`
-6. content `struct { chat-id string, text string }`
-
-Example of a valid encoded payload:
-
-```
-["~#c4",["abc123","text/plain","~:public-group-user-message",154593077368201,1545930773682,["^ ","~:chat-id","testing-adamb","~:text","abc123"]]]
-```
+## Future encoding format
```

Comment:
What is this?

Reply:
Maybe wrong title but you mentioned that in the issue:
> Also considerations re Transit, e.g. if we want to move to a different format like Protobuf. How would compatibility work.

Personally, I would not talk about this here at all.

Reply:
yeah because the obvious answer is it shouldn't work, the current protocol was never intended for backward and forward compatibility, so we either drag this on forever or at some point we'll break compatibility, which will certainly be the case with the new protocol anyway

Reply:
While the current protocol had versioning, it was never used to maintain backward compatibility. Instead v1 of each message has been extended and as result there is no versioning and updates are ad-hoc with newer versions adding older fields to the message to maintain compatibility